Strengthen security & privacy; add WebAppSec coord

wseltzer opened the pull request:
Proposing language from WebGPU charter proposal with stronger security and privacy assurance.

Commit:
Strengthen security & privacy; add WebAppSec coord

Proposing language from WebGPU charter proposal with stronger security and privacy assurance.

jan-ivar commented on charter.html:
```diff
@@ -196,8 +196,8 @@ <h3>Timeline</h3>
 	  <p>In order to advance to <a href="https://www.w3.org/Consortium/Process/#RecsPR" title="Proposed Recommendation">Proposed Recommendation</a>, each specification is expected to have <a href="https://www.w3.org/Consortium/Process/#implementation-experience">at least two independent implementations</a> of each feature defined in the specification.</p>
-	  <p>Each specification should contain a section detailing all known security and privacy implications for implementers, Web authors, and end users.</p>
-	  <p>There should be testing plans for each specification, starting from the earliest drafts.</p>
+<p>Each specification should contain a section detailing any <a href='https://www.w3.org/TR/security-privacy-questionnaire/'>known security</a>, fingerprinting, and privacy implications, and suggested mitigation strategies for implementers, web authors, and end users. The group should not publish a specification if acceptable mitigation strategies cannot be found.</p>	  
```
>  The group should not publish a specification if acceptable mitigation strategies cannot be found.

Acceptable to whom? I'm not against a charter having teeth, just making sure by what metric and how it's going to work. Just making sure we don't set ourselves up if we encounter a problem we cannot solve to everyone's satisfaction.

I couldn't find similar language in webGPU. Do you have a link?

wseltzer replied:
Here's the current draft proposal for WebGPU: https://www.w3.org/2020/06/gpuweb-charter.html

wseltzer replied:
I'd interpret "acceptable" by the usual rules of W3C consensus for Director review, with emphasis that close cases with privacy or security concerns shouldn't default to "just publish a spec."

aboba replied:
Agree with Jan-Ivar's concerns. We've been working on privacy mitigations for webrtc-pc and media-capture for several years, and I don't think we can say that every issue has a deployed mitigation that is approved of by all implementers, web authors and end users, or even that we will have achieved that before reaching Proposed Recommendation.  Instead our goal is to reach WG consensus on solutions, along with transition plans that "don't break the web" in the short term and have a credible shot at wide deployment in the long term.

jan-ivar commented:
Otherwise lgtm. @martinthomson anything you'd like to add?

martinthomson reviewed (commented):
Not sure that fingerprinting needs a special call out next to "privacy implications".

And this has trailing whitespace 😬

That said, I'm comfortable with the idea that we aim to reach consensus whether the risks are properly understood, documented, and mitigated.

dontcallmedom committed:
Remove fingerprinting as special case of privacy implications

also remove trailing whitespace and make link to sec/privacy questionnaire span both topics

dontcallmedom commented:
I've removed the special call-out to fingerprinting (esp as it's unlikely to be a particular concern here) and the trailing white space.

Unless I hear otherwise, I plan to merge this pull request on Monday and proceed with the Advisory Committee review

wseltzer reviewed (commented):
lgtm
